### ISSUE TITLE
Webservice to return binary data

### ISSUE THREAD

**Issue by archerlog** [priore/SOAPEngine#88]
Hi, I am currently having trouble returning binary data from a webservice, all other webservices which are returning string are ok using the SoapEngine however when I am trying to return binary data, I am getting the below error "SOAPEngine missing Content-Length in the header of the HTTP response or its value is equal zero!". I am suspecting this is due to the data returned is too big as compare to return normal strings. 
I have tried both using synchronous and asynchronous and they both have the same problem.

Please help.

The logs I've got is as below:
2015-12-28 12:44:43.130 iPOS[1042:51952] Sended 691 bytes of 691 bytes
2015-12-28 12:44:50.844 iPOS[1042:51952] SOAPEngine missing Content-Length in the header of the HTTP response or its value is equal zero!
2015-12-28 12:44:50.844 iPOS[1042:51952] Received 8838 bytes of 0 bytes
2015-12-28 12:44:50.863 iPOS[1042:51952] Received 17830 bytes of 0 bytes
2015-12-28 12:44:50.887 iPOS[1042:51952] Received 26822 bytes of 0 bytes
2015-12-28 12:44:50.907 iPOS[1042:51952] Received 35813 bytes of 0 bytes
2015-12-28 12:44:50.907 iPOS[1042:51952] Received 35814 bytes of 0 bytes
2015-12-28 12:44:50.952 iPOS[1042:51952] Received 44806 bytes of 0 bytes
2015-12-28 12:44:50.978 iPOS[1042:51952] Received 53798 bytes of 0 bytes
2015-12-28 12:44:50.980 iPOS[1042:51952] Received 62789 bytes of 0 bytes
2015-12-28 12:44:50.995 iPOS[1042:51952] Received 62790 bytes of 0 bytes
2015-12-28 12:44:51.012 iPOS[1042:51952] Received 71782 bytes of 0 bytes
2015-12-28 12:44:51.024 iPOS[1042:51952] Received 80773 bytes of 0 bytes
2015-12-28 12:44:51.025 iPOS[1042:51952] Received 80774 bytes of 0 bytes
2015-12-28 12:44:51.047 iPOS[1042:51952] Received 89766 bytes of 0 bytes
2015-12-28 12:44:51.060 iPOS[1042:51952] Received 98750 bytes of 0 bytes
2015-12-28 12:44:51.073 iPOS[1042:51952] Received 107742 bytes of 0 bytes
2015-12-28 12:44:51.091 iPOS[1042:51952] Received 116734 bytes of 0 bytes
2015-12-28 12:44:51.095 iPOS[1042:51952] Received 125725 bytes of 0 bytes
2015-12-28 12:44:51.130 iPOS[1042:51952] Received 125726 bytes of 0 bytes
2015-12-28 12:44:51.131 iPOS[1042:51952] Received 134718 bytes of 0 bytes
2015-12-28 12:44:51.140 iPOS[1042:51952] Received 143710 bytes of 0 bytes
2015-12-28 12:44:51.161 iPOS[1042:51952] Received 152702 bytes of 0 bytes
2015-12-28 12:44:51.162 iPOS[1042:51952] Received 161693 bytes of 0 bytes
2015-12-28 12:44:51.167 iPOS[1042:51952] Received 161694 bytes of 0 bytes
2015-12-28 12:44:51.171 iPOS[1042:51952] Received 170685 bytes of 0 bytes
2015-12-28 12:44:51.172 iPOS[1042:51952] Received 170686 bytes of 0 bytes
2015-12-28 12:44:51.186 iPOS[1042:51952] Received 179678 bytes of 0 bytes
2015-12-28 12:44:51.214 iPOS[1042:51952] Received 188662 bytes of 0 bytes
2015-12-28 12:44:51.226 iPOS[1042:51952] Received 197654 bytes of 0 bytes
2015-12-28 12:44:51.233 iPOS[1042:51952] Received 206646 bytes of 0 bytes
2015-12-28 12:44:51.264 iPOS[1042:51952] Received 215638 bytes of 0 bytes
2015-12-28 12:44:51.264 iPOS[1042:51952] Received 224629 bytes of 0 bytes
2015-12-28 12:44:51.272 iPOS[1042:51952] Received 224630 bytes of 0 bytes
2015-12-28 12:44:51.277 iPOS[1042:51952] Received 233622 bytes of 0 bytes
2015-12-28 12:44:51.288 iPOS[1042:51952] Received 242614 bytes of 0 bytes
2015-12-28 12:44:51.300 iPOS[1042:51952] Received 251605 bytes of 0 bytes
2015-12-28 12:44:51.316 iPOS[1042:51952] Received 251606 bytes of 0 bytes
2015-12-28 12:44:51.324 iPOS[1042:51952] Received 260598 bytes of 0 bytes
2015-12-28 12:44:51.331 iPOS[1042:51952] Received 269589 bytes of 0 bytes
2015-12-28 12:44:51.340 iPOS[1042:51952] Received 269590 bytes of 0 bytes
2015-12-28 12:44:51.357 iPOS[1042:51952] Received 278574 bytes of 0 bytes
2015-12-28 12:44:51.372 iPOS[1042:51952] Received 287565 bytes of 0 bytes
2015-12-28 12:44:51.376 iPOS[1042:51952] Received 287566 bytes of 0 bytes
2015-12-28 12:44:51.382 iPOS[1042:51952] Received 296558 bytes of 0 bytes
2015-12-28 12:44:51.392 iPOS[1042:51952] Received 305549 bytes of 0 bytes
2015-12-28 12:44:51.415 iPOS[1042:51952] Received 305550 bytes of 0 bytes
2015-12-28 12:44:51.416 iPOS[1042:51952] Received 314542 bytes of 0 bytes
2015-12-28 12:44:51.438 iPOS[1042:51952] Received 323533 bytes of 0 bytes
2015-12-28 12:44:51.441 iPOS[1042:51952] Received 323534 bytes of 0 bytes
2015-12-28 12:44:51.470 iPOS[1042:51952] Received 332526 bytes of 0 bytes
2015-12-28 12:44:51.471 iPOS[1042:51952] Received 341517 bytes of 0 bytes
2015-12-28 12:44:51.476 iPOS[1042:51952] Received 341518 bytes of 0 bytes
2015-12-28 12:44:51.483 iPOS[1042:51952] Received 350510 bytes of 0 bytes
2015-12-28 12:44:51.500 iPOS[1042:51952] Received 359501 bytes of 0 bytes
2015-12-28 12:44:51.523 iPOS[1042:51952] Received 359502 bytes of 0 bytes
2015-12-28 12:44:51.524 iPOS[1042:51952] Received 368494 bytes of 0 bytes
2015-12-28 12:44:51.537 iPOS[1042:51952] Received 377478 bytes of 0 bytes
2015-12-28 12:44:51.556 iPOS[1042:51952] Received 386470 bytes of 0 bytes
2015-12-28 12:44:51.582 iPOS[1042:51952] Received 395462 bytes of 0 bytes
2015-12-28 12:44:51.612 iPOS[1042:51952] Received 404454 bytes of 0 bytes
2015-12-28 12:44:51.661 iPOS[1042:51952] Received 414775 bytes of 0 bytes

**Comment by priore** [comment]
Hello, we need the your code that you are implementing, with all the data required to access the service to try and give you a solution. If it is sensible data, you can send it through support email.

**Comment by archerlog** [comment]
Hi, my sample code to call the webservice is as follows....
However my webservice is currently running locally in my machine and no way to access it remotely as of now.
Basically the webservice will fetch binary data from a db which will be encoded into base64 string when sending via the soap.

soap.userAgent = @"SOAPEngine";
//_soap.licenseKey = @"eJJDzkPK9Xx+p5cOH7w0Q+AvPdgK1fzWWuUpMaYCq3r1mwf36Ocw6dn0+CLjRaOiSjfXaFQBWMi+TxCpxVF/FA==";
    soap.delegate = self;
    soap.version = VERSION_1_1;
    soap.actionNamespaceSlash = NO;
    
    // Generate public token value
    long long tokenTime = [ECMSoapEngine timeInMilliseconds:[NSDate date]];
    NSString *tokenValue = [ECMSoapEngine generateTokenValue:tokenTime];
    
    [soap setValue:app_identifier1 forKey:@"app_identifier_1"];
    [soap setValue:tokenValue forKey:@"app_hash_token"];
    [soap setDoubleValue:tokenTime forKey:@"app_time"];

    // Add additional params here
    for (id key in requests) {
        [soap setValue:[requests objectForKey:key] forKey:key];
    }
    
    // Send synchronous request
    NSError *error = nil;
    [soap syncRequestURL:webServiceUrl soapAction:action error:&error];
    
    if (error) {
        NSLog(@"Soap service error: %@", error.localizedDescription);
        return error;
    }
    
    return [soap stringValue];

**Comment by priore** [comment]
try use this :

``` objective-c
NSDictionary *dict = [soap syncRequestURL:webServiceUrl soapAction:action error:&error];
NSLog("%@", dict); // get value from dictionary
```

**Comment by archerlog** [comment]
Earlier the problem was that the result didn't return anything and it just hangs there with 
2015-12-28 12:44:51.524 iPOS[1042:51952] Received 368494 bytes of 0 bytes
2015-12-28 12:44:51.537 iPOS[1042:51952] Received 377478 bytes of 0 bytes
2015-12-28 12:44:51.556 iPOS[1042:51952] Received 386470 bytes of 0 bytes
2015-12-28 12:44:51.582 iPOS[1042:51952] Received 395462 bytes of 0 bytes
2015-12-28 12:44:51.612 iPOS[1042:51952] Received 404454 bytes of 0 bytes
2015-12-28 12:44:51.661 iPOS[1042:51952] Received 414775 bytes of 0 bytes

I realised that I didn't set the responseHeader for the soap hence since I've added the soap.responseHeader = YES;
I've managed to get the return result. 

But the error message is still there.
2015-12-30 10:59:27.446 iPOS[676:18576] SOAPEngine missing Content-Length in the header of the HTTP response or its value is equal zero!
2015-12-30 10:59:27.446 iPOS[676:18576] Received 8838 bytes of 0 bytes
2015-12-30 10:59:27.452 iPOS[676:18576] Received 17830 bytes of 0 bytes
2015-12-30 10:59:27.464 iPOS[676:18576] Received 26822 bytes of 0 bytes

Wondering is this a bug or something else?
